MammoNet/dataset/data_handler.py

```diff
@@ -66,65 +68,67 @@ def create_stratified_datasets(self, file_paths, labels, sublabels, resolutions,
-        '''
+
+    def generate_augmented_images(self, input_images_paths, input_labels, num_copies=1):
+        """
         Generate augmented images for the input images.
-        '''
+        """
         augmented_images = []
         augmented_labels = []
-        
+
         if self.augment:
-            if self.reuse_augmentation is False:
-                for idx, (label, input_image_path) in tqdm(enumerate(zip(input_labels, input_images_paths)), total=len(input_labels), desc="Augmenting images"):
+            if not self.reuse_augmentation:
+                for idx, (label, input_image_path) in tqdm(enumerate(zip(input_labels, input_images_paths)),
+                                                           total=len(input_labels), desc="Augmenting images"):
                     img = Image.open(input_image_path)
                     img_array = np.array(img)
-            
+
                     augmenter = ImageAugmentations()
-                    
+
                     for i in range(num_copies):
                         augmented_image = augmenter(img_array)  # Apply augmentations
                         augmented_img_pil = augmented_image
-                            
+
                         output_path = os.path.join(self.augmentation_dir, f"augmented_{idx}_{label}.png")
-                        
+
                         augmented_img_pil.save(output_path)
-                        
+
                         augmented_images.append(output_path)
                         augmented_labels.append(label)
             else:
-                augmented_images = [os.path.join(self.augmentation_dir, file) 
-                                    for file in os.listdir(self.augmentation_dir) 
+                augmented_images = [os.path.join(self.augmentation_dir, file)
+                                    for file in os.listdir(self.augmentation_dir)
                                     if file.endswith('.png')]
-                augmented_labels = [get_label_from_augmented_image_path(file) 
+                augmented_labels = [get_label_from_augmented_image_path(file)
                                     for file in augmented_images]
-                
+
         return augmented_images, augmented_labels
 
-    def create_datasets_with_augmentation(self, train_files, val_files, test_files, train_labels, val_labels, test_labels, augment = True):
+    def create_datasets_with_augmentation(self, train_files, val_files, test_files, train_labels, val_labels,
+                                          test_labels):
         """
         Create datasets with augmentation for training, validation, and testing.
         """
-        
-        augmented_files, augmented_labels = self.generate_augmented_images(train_files, train_labels)
-        train_files = train_files + augmented_files
-        train_labels = tuple(list(train_labels) + augmented_labels)
-        
+        if self.augment:
```

Comment:
Hi! Augmentation is passed on in the constructor and generate_augmented_images already has an if clause for augmentation and reuse of augmentation, so this if is redundant as far as I see